defenz_alerts/main.py

```diff
@@ -0,0 +1,484 @@
+
+    if not args.report_email:
+        raise Exception(error_temp.format('Report Email'))
+
+
+def get_command_line_arguments():
+    """
+    Gets command line arguments.
+
+    :return: arguments
+    """
+    args_parser = \
+        argparse.ArgumentParser(
+            description="""This script is used to send scheduled Defenz alerts
+            by using Defenz APIs.""")
+
+    args_parser.add_argument(
+        '-u',
+        '--username',
+        default=CONFIG['DEFENZ']['USERNAME'],
+        help='Defenz portal username. Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-p',
+        '--password',
+        default=CONFIG['DEFENZ']['PASSWORD'],
+        help='Defenz portal password. Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-c',
+        '--client-id',
+        default=CONFIG['DEFENZ']['CLIENT_ID'],
+        help='Defenz client id (customer name). Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-s',
+        '--client-secret',
+        default=CONFIG['DEFENZ']['CLIENT_SECRET'],
+        help='Defenz client secret. Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-e',
+        '--report-email',
+        default=CONFIG['DEFENZ']['REPORT_EMAIL'],
+        help='The report will be sent to this email address '
+             'if network report email is not set. '
+             'Default: read from config file'
+    )
+
+    args_parser.add_argument(
+        '-n',
+        '--network-ids',
+        nargs='*',
+        help='List of network ids. Default all networks'
+    )
+
+    args_parser.add_argument(
+        '-r',
+        '--report-types',
+        nargs='*',
+        default=[MALWARE_PHISHING_BLOCKS, BOT_NET_BLOCKS],
+        choices=[MALWARE_PHISHING_BLOCKS, BOT_NET_BLOCKS, WEB_FILTER_BLOCKS],
+        help='List of report types. Default {} and {}'.format(
+            MALWARE_PHISHING_BLOCKS, BOT_NET_BLOCKS)
+    )
+
+    args_parser.add_argument(
+        '-i',
+        '--interval',
+        default=5,
+        help='Report interval in minute. Default is 5 minutes'
+    )
+
+    args_parser.add_argument(
+        '-v',
+        '--verbose',
+        default=False,
+        action='store_true',
+        help='Write the logs in console as well. Otherwise the logs only will '
+             'be written in the log files. The log files are located in '
+             '$HOME/defenz'
+    )
+
+    return args_parser.parse_args()
+
+
+def main():
+    """
+    The main function of the script.
+    """
+
+    try:
+        args = get_command_line_arguments()
+
+        if args.verbose:
+            console_handler = logging.StreamHandler()
+            console_handler.setLevel(logging.DEBUG)
+            console_handler.setFormatter(
+                logging.Formatter('%(asctime)s - %(levelname)s: %(message)s'))
+            LOGGER.addHandler(console_handler)
+
+        validate_arguments(args)
```

Comment:
I can see that arguments validation is failing silently when not in verbose mode. 

It may be a good idea that validation writes to standard output (along with log, as it is already doing) regardless of the defined verbose option to quickly inform the user that there is a problem on arguments. What do you think?

Reply:
The point of having quite/silent mode is to use when you want to run it with a scheduler. in that case you have to look into the logs to see if there is any error.
I get your point but that make it a bot complicated. We have to have a streamhander only for argument validation our use print.
I think it make sense to run the script in verbose mode once before setting in scheduler to make sure it's working fine. I suggest we keep it as is but we can log an issue to improve it maybe? But if you think it's necessary I can think to find a solution now.

Reply:
That is not a blocker at all. 

But it would easy the first use of the script, which is very important when we want to engage the collaboration. We definitely can write it down as an improvement, but it would be very handy if we have it on the first release.

Verbose does what we want, but the thing here is that we rely on users' choice or experience to run verbose mode for the first time, what he may not do it for various reasons.

I was thinking that we could raise a `RequirementeException` to catch it when called and, with that, add the StreamHandler as it verbose mode was provided.

We could put on the top of the backlog for now if we do not have time to implement. What do you think?